src/OpenTelemetry.Collector.Azure/AzureSdkCollector.cs

```diff
@@ -0,0 +1,150 @@
+        {
+            this.tracer = tracer;
+            this.sampler = sampler;
+
+            this.subscriptions.Add(DiagnosticListener.AllListeners.Subscribe(this));
+        }
+
+        public void Dispose()
+        {
+            lock (this.subscriptions)
+            {
+                foreach (var subscription in this.subscriptions)
+                {
+                    subscription.Dispose();
+                }
+            }
+        }
+
+        public void OnCompleted()
+        {
+        }
+
+        public void OnError(Exception error)
+        {
+        }
+
+        public void OnNext(KeyValuePair<string, object> value)
+        {
+            if (value.Key.EndsWith("Start"))
+            {
+                this.OnStartActivity(Activity.Current, value.Value);
+            }
+            else if (value.Key.EndsWith("Stop"))
+            {
+                // Current.Parent is used because OT wraps additional Activity over
+                this.OnStopActivity(Activity.Current.Parent, value.Value);
+            }
+            else if (value.Key.EndsWith("Exception"))
+            {
+                // Current.Parent is used because OT wraps additional Activity over
+                this.OnException(Activity.Current.Parent, value.Value);
+            }
+        }
+
+        public void OnNext(DiagnosticListener value)
+        {
+            if (value.Name.StartsWith("Azure"))
+            {
+                lock (this.subscriptions)
+                {
+                    this.subscriptions.Add(value.Subscribe(this));
+                }
+            }
+        }
+
+        private void OnStartActivity(Activity current, object valueValue)
+        {
+            var operationName = current.OperationName;
+            foreach (var keyValuePair in current.Tags)
+            {
+                if (keyValuePair.Key == "http.url")
+                {
+                    operationName = keyValuePair.Value;
+                }
+            }
+
+            var span = this.tracer.SpanBuilder(operationName)
+                .SetSampler(this.sampler)
+                .StartSpan();
+
+            this.scopes.TryAdd(current, this.tracer.WithSpan(span));
+        }
+
+        private void OnStopActivity(Activity current, object valueValue)
+        {
+            var span = this.tracer.CurrentSpan;
+            foreach (var keyValuePair in current.Tags)
+            {
+                span.SetAttribute(keyValuePair.Key, keyValuePair.Value);
+            }
+
+            this.scopes.TryRemove(current, out var scope);
+
+            scope?.Dispose();
+        }
+
+        private void OnException(Activity current, object valueValue)
+        {
+            var span = this.tracer.CurrentSpan;
+            foreach (var keyValuePair in current.Tags)
+            {
+                span.SetAttribute(keyValuePair.Key, keyValuePair.Value);
+            }
+
+            span.Status = Status.Unknown;
```

Comment:
it doen't really make sense to add tag AND set status. You can put exception info into status with `Status.Unknown.WithDesctiption(kvp.Value)`

Reply:
Why? There might be more useful information in tags.

Reply:
wait, you add tags in start AND stop  AND exception. why? Adding them in stop is enough and you likely triplicating them now.  Also tags and status are really independent. 

Status should contain info about the exception. Status (if an exception happened) should contain exception info: i.e. you'd need to translate the exception into the Status and also add description to the status.